File: 2843-count-symmetric-integers/2843-count-symmetric-integers.cpp
class Solution {
public:
    int countSymmetricIntegers(int low, int high) {
        int c = 0;
        for(int x = low ; x <= high ; x++ ){
            string s = to_string(x);
            int l = 0 , r = 0;
            int n = s.size();
            if(n % 2 != 0)  continue;
            for(int i = 0 ; i < n ; i++){
                if(i < n /2){
                    l += (s[i] - '0');
                }
                else{
                r += (s[i] - '0');
                }
            }
            if(l == r){
                c++;
            }
        }
        return c;
    }
};

File: 2843-count-symmetric-integers/2843-count-symmetric-integers.java
class Solution {
    public int countSymmetricIntegers(int low, int high) {
        int c = 0;
        for(int x = low ; x <= high ; x++){
            String str = String.valueOf(x);
            int n = str.length();
            if(n % 2 != 0)  continue;
            int l = 0 , r = 0;
            for(int d = 0 ; d < n ; d++){
                if(d < n/2){
                    l += (str.charAt(d) - '0');
                }
                else{
                    r += (str.charAt(d) - '0');
                }
            }
            if (l == r){
                c++;
            }
        }
        return c;
    }
}